go/vt/automation/vtctl_task.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2015, Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package automation
+
+import (
+	"bytes"
+	"os/exec"
+	"strings"
+
+	pb "github.com/youtube/vitess/go/vt/proto/automation"
+)
+
+// VtctlTask executes known vtctl commands. Output is preserved.
+type VtctlTask struct {
+}
+
+func (t *VtctlTask) run(parameters map[string]string) ([]*pb.TaskContainer, string, error) {
+	// TODO(mberlin): Use the vtctl code directly.
```

Comment:
Yeah this is not gonna work as is, vtctl would need a lot of extra command line parameters to work. You can use vtctlclient code, or link with vtctl library (the scheduler will end up being in vtctld anyway).